src/tests/string/test_string_functions.f90

```diff
@@ -52,6 +53,46 @@ subroutine test_reverse_string
 
     end subroutine test_reverse_string
 
+    subroutine test_slice_string
+        type(string_type) :: test_string
+        character(len=:), allocatable :: test_char
+        test_string = "abcdefghijklmnopqrstuvwxyz"
+        test_char = "abcdefghijklmnopqrstuvwxyz"
+
+        call check(slice(test_string, 2, 16, 3) == "behkn", &
+                    'function slice failed', warn=.true.)
```

Comment:
Be careful when using `warn=.true.` in the tests, as a failing test will only produce output but don't change the exit status of the test driver. A failing test wouldn't show up in the CI with `warn=.true.`